packages/angular-test/src/range.ts

```diff
@@ -0,0 +1,382 @@
+                        schema: testData.schema,
+                    }
+                }
+            });
+            mockSubStore.complete();
+            fixture.detectChanges();
+            component.ngOnInit();
+            expect(rangeElement.nativeElement.id).toBe('myId');
+
+        });
+    };
+export const rangeInputEventTest = <C extends JsonFormsControl, I>(
+    testConfig: TestConfig<C>,
+    instance: Type<I>) => () => {
+        let fixture: ComponentFixture<any>;
+        let rangeNativeElement: any;
+        let component: C;
+
+        baseSetup(testConfig);
+
+        beforeEach(() => {
+            const preparedComponents = prepareComponent(testConfig, instance);
+            fixture = preparedComponents.fixture;
+            rangeNativeElement = preparedComponents.rangeElement.nativeElement;
+            component = preparedComponents.component;
+        });
+
+        it('should update via input event', () => {
+
+            const mockSubStore = MockNgRedux.getSelectorStub();
+            component.uischema = testData.uischema;
+            component.schema = testData.schema;
+
+            mockSubStore.next({
+                jsonforms: {
+                    core: {
+                        data: testData.data,
+                        schema: testData.schema,
+                    }
+                }
+            });
+            mockSubStore.complete();
+            fixture.detectChanges();
+            component.ngOnInit();
+
+            const spy = spyOn(component, 'onChange');
+            const trackElement = rangeNativeElement.querySelector('.mat-slider-wrapper');
+            const dimensions = trackElement.getBoundingClientRect();
+            const x = dimensions.left + (dimensions.width * 0.2);
+            const y = dimensions.top + (dimensions.height * 0.2);
+            const event = document.createEvent('MouseEvent');
+
+            event.initMouseEvent('click',
+                                 true, /* canBubble */
+                                 false, /* cancelable */
+                                 window, /* view */
+                                 0, /* detail */
+                                 x, /* screenX */
+                                 y, /* screenY */
+                                 x, /* clientX */
+                                 y, /* clientY */
+                                 false, /* ctrlKey */
+                                 false, /* altKey */
+                                 false, /* shiftKey */
+                                 false, /* metaKey */
+                                 0, /* button */
+                                 null /* relatedTarget */);
+            // `initMouseEvent` doesn't allow us to pass the `buttons` and
+            // defaults it to 0 which looks like a fake event.
+            Object.defineProperty(event, 'buttons', { get: () => 1 });
+            rangeNativeElement.dispatchEvent(event);
```

Comment:
Could this be simplified as in `fixture.debugElement.query(By.css('.mat-slider-wrapper')).nativeElement.click()` ?

Reply:
I will try this out and comment with the results